components/section-render.js

```diff
@@ -0,0 +1,159 @@
+      the order of the sections in the URL (section-render-url).
+      For example: 
+      section-render-trigger="custom-content,custom-content-2" 
+      should correspond to 
+      section-render-url="?sections=contact-form,featured-blog", 
+      meaning "custom-content" will render "contact-form" and "custom-content-2" will render "featured-blog".
+    */
+}
+
+class SectionRender extends HTMLElement {
+  constructor() {
+    super();
+    this.renderTriggerAttr = "section-render-trigger";
+    this.renderTargetAttr = "section-render-content";
+    this.renderUrlAttr = "section-render-url";
+
+    this.triggerButtons = this.querySelectorAll(`[${this.renderTriggerAttr}]`);
+    this.url = null;
+  }
+
+  connectedCallback() {
+    this.triggerButtons.forEach((el) => {
+      el.addEventListener("click", (e) => {
+        e.preventDefault();
+
+        let sectionId = el.getAttribute(this.renderTriggerAttr);
+        this.url = el.getAttribute(this.renderUrlAttr);
+
+        if (sectionId.includes(",")) {
+          sectionId = sectionId.split(",");
+        }
+
+        this.updateSection(sectionId);
+      });
+    });
+  }
+
+  async updateSection(sectionIds) {
+    let sections = null;
+
+    if (Array.isArray(sectionIds)) {
+      sections = sectionIds
+        .map((id) =>
+          Array.from(this.querySelectorAll(`[${this.renderTargetAttr}=${id}]`))
+        )
+        .flat();
+    } else {
+      sections = this.querySelectorAll(
+        `[${this.renderTargetAttr}=${sectionIds}]`
+      );
+    }
+
+    try {
+      const res = await fetch(this.url);
+
+      if (!res.ok) {
+        throw new Error(`Load error: ${res.status} ${res.statusText}`);
+      }
+
+      const responseData = await res.text();
+
+      let contentContainer = document.createElement("div");
+
+      if (this.isJsonObject(responseData)) {
+        this.renderMultipleContents(
+          sections,
+          responseData,
+          sectionIds,
+          contentContainer
+        );
+      } else {
+        this.renderSingleContent(sections, responseData, contentContainer);
+      }
+    } catch (error) {
+      console.error("Error on section loading:", error);
+    }
+  }
+
+  renderMultipleContents(sections, response, sectionIds, contentContainer) {
+    const parsedResponse = JSON.parse(response);
+    const parsedResponseKeys = Object.keys(parsedResponse);
+
+    const sectionsMarkup = sectionIds.reduce(
+      (accumulator, sectionId, index) => {
+        const sectionName = parsedResponseKeys[index];
+
+        if (sectionName) {
+          accumulator[sectionId] = parsedResponse[sectionName];
+        }
+        return accumulator;
+      },
+      {}
+    );
+
+    sections.forEach((section) => {
+      const contentId = section.getAttribute(this.renderTargetAttr);
+
+      if (sectionsMarkup[contentId]) {
+        contentContainer.innerHTML = sectionsMarkup[contentId];
+
+        section.innerHTML = contentContainer.innerHTML;
```

Comment:
I'm noticing that when multiple contents are loaded, there is a 3x flicking for the 3 pieces of content. I'm wondering if the fact that there are two writes to `.innerHTML` is causing this to happen. Look into why this is happening.

https://github.com/user-attachments/assets/2420c0b8-6bc1-4be1-9010-20559a514e1c

Reply:
Oh... my fault, I checked it out. The problem is that I wrapped section-render tag 2 times. You can see that on click I have 2 requests because of that too.

Here is image:
![image](https://github.com/user-attachments/assets/95ed09b9-f019-4bdc-ab78-79a0adc0952e)

Updated: https://bekzod-test.myshopify.com/?preview_theme_id=140667846824
Password: ahlyeu

 @darcyvoutt